src/LogicTasks/Syntax/IllegalFormulas.hs

```diff
@@ -84,19 +84,11 @@ completeGrade
   => FilePath
   -> LegalPropositionInst
   -> [Int]
-  -> LangM m
-completeGrade path inst sol = refuseIfWrong $ do
-  when wrongSolution $ do
-     instruct $ do
-        english "Your solution is incorrect."
-        german "Ihre Lösung ist falsch."
-
-  when (showSolution inst) $ do
-    when wrongSolution $
-      example (show serialsOfWrong) $ do
-          english "A possible solution for this task is:"
-          german "Eine mögliche Lösung für die Aufgabe ist:"
+  -> Rated m
+completeGrade path LegalPropositionInst{..} sol = do
+  x <- multipleChoice IndefiniteArticle what solutionDisplay solution sol
```

Comment:
Is the behavior correct here? I'm not sure what happens when `multipleChoice` does `refuse`. Will the remainder of the `do`-Block here even be executed?

Reply:
Maybe use of this function would be appropriate here: https://github.com/fmidue/modelling-tasks/blob/b5dbd4be1b9be0e25c55ed8f8cea5755621376ef/src/Modelling/Auxiliary/Output.hs#L85-L93 as exemplified in https://github.com/fmidue/modelling-tasks/blob/b5dbd4be1b9be0e25c55ed8f8cea5755621376ef/src/Modelling/CdOd/RepairCd.hs#L358-L363

Reply:
It wasn't shown, actually. `multipleChoice` is still a bit confusing for me. Why doesn't it refuse (or accept with 0 points) when the solution list is empty, even tough it should have entries?

Reply:
Well, in a task with four possible answers, of which two are correct, a student claiming that all are wrong still should get 0.5 points, because they got two of the four judgements correct.

"Which of the following four properties are true of the number 13: a) is odd, b) is even, c) is prime, d) is non-prime? Answer: []. Half points. Answer: [b,d]. Zero points."

Reply:
That makes sense. I'll remove the checks I added then.